src/components/modules/crossBlockSelection.ts

```diff
@@ -173,6 +204,17 @@ export default class CrossBlockSelection extends Module {
     this.lastSelectedBlock = targetBlock;
   }
 
+  /**
+   * Removes all ranges when any Block is selected
+   */
+  private onSelectionChange = (): void => {
+    const { BlockSelection } = this.Editor;
+
+    if (BlockSelection.anyBlockSelected) {
```

Comment:
it will iterate through all the blocks on almost every mousemove firing. Maybe we can optimize it?